README.md

```diff
@@ -48,15 +48,26 @@ We should write an integration test for each new user interaction we add to our
 
 ### How do we do it ?
 
-- First, find out the different api calls that you'll have to mock for your test.
-  To do that, open up your simulator along with your debugger. Then execute the feature you want to test and check the api calls that occured.
-- Then find out how to organise your mock data
-- Finally write your tests following the model
-  - SETUP --> mock api calls, render page
+We can separate our tests into 4 different parts:
+
+  - SETUP --> api calls mocks, page rendering
   - GIVEN --> what the user sees on the page
   - WHEN --> what the user does
   - THEN --> what visual feedback the user should expect
 
+When you want to write a test, you have to ask the questions in the reverse order:
+
+  - THEN -> the end result we want to protect with our test
+    - e.g. I want a modal to show up, I want to be redirected to another page, I want my snapshot to match, I want an amount to be displayed
+  - WHEN -> the furthest action you can easily trigger
+    - e.g. I click on a button, I enter some text, I dispatch a Redux action, I dispatch a native event
```

Comment:
I'm ok for all of them except the dispatch of a redux action (I don't see a use case where a user interaction / native event couldn't do the job and trigger the redux action, i see it more as an implementation detail)

Reply:
One example we had on our project:

- we have a modal at the root of our App (same level as the RootNavigator)
- this modal is controlled by Redux
- actions to display the modal are dispatched inside pages inside the RootNavigator

If we wanted to test with a user Interaction, we saw no other way than mounting the whole app for the test.
Actually, I believe we could have mounted one real button (that dispatches the action) and the modal, and then trigger it by user interaction. Is it something you have already done?

Reply:
Ok just saw how to do it in the docs, I updated my file :)